dom.bs

```diff
@@ -4176,11 +4167,13 @@ when invoked, must run these steps:
   JavaScript implementations a cached <code class='lang-javascript'>Math.random()</code> value can
   be used.
 
- <li><p>If <var>node1</var> is an <a>ancestor</a> of <var>node2</var>, then return the result of
- adding {{Node/DOCUMENT_POSITION_CONTAINS}} to {{Node/DOCUMENT_POSITION_PRECEDING}}.
+ <li><p>If <var>node1</var> is an <a>ancestor</a> of <var>node2</var>, or <var>node1</var> is
+ <var>node2</var> and <var>attr2</var> is non-null, then return the result of adding
+ {{Node/DOCUMENT_POSITION_CONTAINS}} to {{Node/DOCUMENT_POSITION_PRECEDING}}.
 
- <li><p>If <var>node1</var> is a <a>descendant</a> of <var>node2</var>, then return the result of
- adding {{Node/DOCUMENT_POSITION_CONTAINED_BY}} to {{Node/DOCUMENT_POSITION_FOLLOWING}}.
+ <li><p>If <var>node1</var> is a <a>descendant</a> of <var>node2</var>, or <var>node1</var> is
+ <var>node2</var> and <var>attr1</var> is non-null, then return the result of adding
+ {{Node/DOCUMENT_POSITION_CONTAINED_BY}} to {{Node/DOCUMENT_POSITION_FOLLOWING}}.
 
  <li><p>If <var>node1</var> is <a>preceding</a> <var>node2</var>, then return
```

Comment:
A  note around here that explains that we can end up here for attributes and that they are considered to come before the real descendants would be nice, since it's going to be impossible to extract intention from the algorithm alone.